WooCommerce/Classes/POS/Controllers/PointOfSaleCouponsController.swift

```diff
@@ -64,20 +66,47 @@ import protocol Yosemite.PointOfSaleCouponServiceProtocol
 
 @available(iOS 17.0, *)
 private extension PointOfSaleCouponsController {
+    /// Loads the first page by attempting to load the first page from local storage
+    /// then syncs from the remote regardless the result
+    func loadFirstPage() async {
+        do {
+            let storedCoupons = try await couponProvider.provideLocalPointOfSaleCoupons()
+
+            if !storedCoupons.isEmpty {
+                setCouponsLoadedViewState(storedCoupons, hasMoreItems: true)
+            }
+            let coupons = try await couponProvider.providePointOfSaleCoupons(pageNumber: 1)
+            if !coupons.items.isEmpty {
+                setCouponsLoadedViewState(coupons.items, hasMoreItems: true)
+            } else {
+                setCouponsEmptyViewState()
+            }
```

Comment:
Could we reuse `fetchCoupons` here?
            
```suggestion
            try await fetchCoupons(pageNumber: 1)
```
```

Reply:
We can! https://github.com/woocommerce/woocommerce-ios/pull/15499/commits/51debf7de6f3a3bc9d687cc6f2bc74e49eb59a87